src/zenml/integrations/aws/service_connectors/aws_service_connector.py

```diff
@@ -713,7 +713,10 @@ def get_boto3_session(
             # Refresh expired sessions
             now = datetime.datetime.now(datetime.timezone.utc)
             expires_at = expires_at.replace(tzinfo=datetime.timezone.utc)
-            if expires_at > now:
+            # check if the token expires in the next 10 minutes
```

Comment:
```suggestion
            # check if the token expires in the next 5 minutes
```

This or change the timedelta :)

I can't see any issues with this change but I'm also not super familiar with service connectors. Worst case this would lead to refreshing the session more often, correct? Is this something we would also need in other service connector implementations?

Reply:
- I'll make the change to 5. thanks for the catch :)
- a session is valid for 1 hour and so it shouldn't lead to frequent refreshing of session creds.
- we already have a buffer when refreshing tokens for the EKS cluster (which we get using the session) so there is precedent. I remember we did this because someone ran into this edge case on slack. This change here is just precautionary and there have been no reports yet.
- I'll take a look at other service connector implementations to check if this is needed elsewhere.

Reply:
Alright sounds good 👌